### PR TITLE
Add CUB_ENABLE_CPP_DIALECT_IN_NAMES CMake flag

### PR DESCRIPTION
**Pull request by allisonvacanti**
On by default. When disabled:
a) Emit configuration error if more than one dialect option is enabled, and
b) Remove dialect information from target / executable names.

Fixes NVIDIA/cub#250.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,6 +54,9 @@ option(CUB_ENABLE_THOROUGH_TESTING "Build CUB thorough test variants." ON)
 option(CUB_ENABLE_BENCHMARK_TESTING  "Build CUB benchmark test variants." ON)
 option(CUB_ENABLE_MINIMAL_TESTING "Build CUB minimal test variants." ON)
 option(CUB_ENABLE_EXAMPLES "Build CUB examples." ON)
+option(CUB_NVCXX_QA_COMPAT "Use shorter names and force a single dialect." OFF)
```
Comment (brycelelbach) [inline CMakeLists.txt:57]: I'd prefer a name that describes what it does, not what it is. I'd suggest something like `CUB_ENABLE_CPP_DIALECT_IN_EXE_NAMES` (defaulted to `ON`). I don't think the name needs to capture the fact that you get a configuration error if you try to use it with multiple C++ dialects; that just seems like a natural consequence.

**Review by brycelelbach — CHANGES_REQUESTED** [review]
Looks good, but I'm not thrilled with the name. I'd prefer something that describes what it does, not what it's used for.

**Comment by allisonvacanti** [comment]
@brycelelbach Updated with requested changes.

run tests

**Review by brycelelbach — APPROVED** [review]
Approved, please just fix the comment I pointed out. Let me know when this is merged so I can add CUB tests to NVC++'s internal testing!

**Pull request merged**